src/dune_lang/syntax.ml

```diff
@@ -44,22 +44,79 @@ module Version = struct
 end
 
 module Supported_versions = struct
-  type t = int Int.Map.t
-
-  let to_dyn = Int.Map.to_dyn Int.to_dyn
-
-  let make = Int.Map.of_list_exn
-
-  let greatest_supported_version t = Option.value_exn (Int.Map.max_binding t)
-
-  let is_supported t (major, minor) =
+  (* The extension supported versions are declared using an explicit list of all
+     versions but stored as a map from major versions to maps from minor version
+     to dune_lang required versions *)
+  type t = Version.t Int.Map.t Int.Map.t
+
+  let to_dyn t = Int.Map.to_dyn (Int.Map.to_dyn Version.to_dyn) t
+
+  (* We convert the exposed extension version type: `(Version.t * [ `Since of
+     Version.t ]) list` which is a list of fully qualified versions paired with
```

Comment:
BTW, we generally use the odoc syntax rather than markdown even in simple comments. I changed this one as I was adding an example in the doc.